ml-agents-envs/mlagents_envs/rpc_utils.py

```diff
@@ -45,6 +45,35 @@ def behavior_spec_from_proto(
     return BehaviorSpec(observation_shape, action_type, action_shape)
 
 
+class OffsetBytesIO:
+    """
+    Simple file-like class that wraps a bytes, and allows moving its "start"
+    position in the bytes. This is only used for reading concatenated PNGs,
+    because Pillow always calls seek(0) at the start of reading.
+    """
+
+    __slots__ = ["fp", "offset"]
+
+    def __init__(self, data: bytes):
+        self.fp = io.BytesIO(data)
+        self.offset = 0
+
+    def seek(self, offset: int, whence: int = io.SEEK_SET) -> int:
+        if whence == io.SEEK_SET:
+            res = self.fp.seek(offset + self.offset)
+            return res - self.offset
+        raise NotImplementedError()
+
+    def tell(self) -> int:
+        return self.fp.tell() - self.offset
+
+    def read(self, size: int = -1) -> bytes:
+        return self.fp.read(size)
+
+    def _real_tell(self) -> int:
```

Comment:
Seems odd for this to be `_real_tell` when it's not used privately / internally.

Reply:
Yeah, wasn't sure what to call it. Do you think I should remove it and just use `fp.tell` directly?

Reply:
Oh, I was just commenting on the initial underscore indicating it is a "private" method. I think "real_tell" is fine, or maybe "force_tell".